File: Challange/MAY/2.4.py
class Solution:
    def dfs(self, image,i : int,j : int,color : int, fill : int):
        # call recussion!
        if( i < 0 or j < 0 or i >= len(image) or j >= len(image[0]) or image[i][j] != color):
            return
        image[i][j] = fill
        self.dfs(image,i+1,j,color,fill)
        self.dfs(image,i-1,j,color,fill)
        self.dfs(image,i,j+1,color,fill)
        self.dfs(image,i,j-1,color,fill)
        
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        if(image[sr][sc] == newColor):
            return image
        self.dfs(image,sr,sc,image[sc][sr], newColor)
        return image

File: Challange/MAY/2.4.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution
{
public:
    void dfs(vector<vector<int>> &image, int ci, int cj, int newVal, int matchVal)
    {
        // out of bound check!
        if (ci < 0 || cj < 0 || ci >= image.size() || cj >= image[0].size() || image[ci][cj] != matchVal)
            return;
        image[ci][cj] = newVal;
        dfs(image, ci + 1, cj, newVal, matchVal);
        dfs(image, ci - 1, cj, newVal, matchVal);
        dfs(image, ci, cj + 1, newVal, matchVal);
        dfs(image, ci, cj - 1, newVal, matchVal);
    }
    vector<vector<int>> floodFill(vector<vector<int>> &image, int sr, int sc, int newColor)
    {
        if (image[sr][sc] == newColor)
            return image;
        dfs(image, sr, sc, newColor, image[sr][sc]);
        return image;
    }
};